daemon/src/maker_cfd.rs

```diff
@@ -193,35 +217,50 @@ impl Actor {
         Ok(())
     }
 
-    async fn handle_propose_settlement(&mut self, proposal: SettlementProposal) -> Result<()> {
+    async fn handle_propose_settlement(
+        &mut self,
+        taker_id: TakerId,
+        proposal: SettlementProposal,
+    ) -> Result<()> {
         tracing::info!(
             "Received settlement proposal from the taker: {:?}",
             proposal
         );
         self.current_pending_proposals.insert(
             proposal.order_id,
-            UpdateCfdProposal::Settlement {
-                proposal,
-                direction: SettlementKind::Incoming,
-            },
+            (
+                UpdateCfdProposal::Settlement {
+                    proposal,
+                    direction: SettlementKind::Incoming,
+                },
+                taker_id,
```

Comment:
We solved it slightly different and stored the `taker_id` inside the `SettlementKind::Incoming`. But I guess your approach is also nice :)